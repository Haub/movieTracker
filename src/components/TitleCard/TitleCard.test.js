import React from 'react';
import { shallow } from 'enzyme';
import { TitleCard } from './index.js';
import { mapStateToProps, mapDispatchToProps } from './index.js';


describe('TITLE CARD', () => {
  let controlFavorites;
  let mockUser;
  let wrapper;
  
  beforeEach(() => {
    controlFavorites = jest.fn();
    window.alert = jest.fn();
    mockUser = {id: 2, name: 'Graham', email: 'graham@aol.com', password: '123', favorites: ['Venom']};
    wrapper = shallow(
      <TitleCard 
        user={mockUser}
        video={[{key: true}, {key: true}] } 
        mpaa={{certification: 'true'}} 
        overview={'hello'}
        controlFavorites={controlFavorites}
      />);
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have an initial state with a key of play set to false', () => {
    expect(wrapper.state().play).toEqual(false);
  });

  it('should set the state of play to true when mouseEnter is invoked', () => {
    expect(wrapper.state().play).toEqual(false);
    wrapper.instance().mouseEnter();
    expect(wrapper.state().play).toEqual(true);
  });

  it('should set the state of play to false when mouseLeave is invoked', () => {
    wrapper.instance().mouseLeave();
    expect(wrapper.state().play).toEqual(false);
  })

  it('should send an alert if no user id exists', () => {
    mockUser = {};
    wrapper = shallow(
      <TitleCard 
        user={mockUser}
        video={[{key: true}, {key: true}] } 
        mpaa={{certification: 'true'}} 
        overview={'hello'}
        controlFavorites={controlFavorites}
      />);
    wrapper.instance().toggleFavorite();
    expect(window.alert).toHaveBeenCalled();

  })
})

describe('mapStateToProps', () => {
  it('should return a user object', () => {
    const mockState = {
      user: {id: 2, name: 'Graham', email: 'graham@aol.com', password: '123', favorites: ['Venom']}
    }
    const expected = {
      user: {id: 2, name: 'Graham', email: 'graham@aol.com', password: '123', favorites: ['Venom']}
    }
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });
})

describe('mapDispatchToProps', () => {
  it('calls dispatch with a controlFavorites action when toggleFavorites is invoked', () => {
    const mockDispatch = jest.fn();
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.toggleFavorite();
    expect(mockDispatch).toHaveBeenCalled();
  })
  
})

