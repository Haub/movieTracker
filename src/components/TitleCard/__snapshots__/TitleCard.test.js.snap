// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TITLE CARD should match the snapshot 1`] = `
<div
  className="title-card fade"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="foreground"
  >
    <div
      className="info-section"
    >
      <div />
    </div>
    <div
      className="favorite-section"
    />
  </div>
  <VideoPlayer
    feature={false}
    play={false}
    url={true}
  />
</div>
`;
